lib/auth/auth_with_roles.go

```diff
@@ -4018,7 +4040,8 @@ func (a *ServerWithRoles) validateRole(ctx context.Context, role types.Role) err
 
 // CreateRole creates a new role.
 func (a *ServerWithRoles) CreateRole(ctx context.Context, role types.Role) (types.Role, error) {
-	if err := a.action(apidefaults.Namespace, types.KindRole, types.VerbCreate); err != nil {
+	sctx := &services.Context{User: a.context.User, Resource: role}
+	if err := a.actionWithContext(sctx, apidefaults.Namespace, types.KindRole, types.VerbCreate); err != nil {
```

Comment:
Similar to my question above, what does changing this from "action" to "actionWithContext" accomplish?